code/game/turfs/turf.dm

```diff
@@ -186,11 +187,14 @@
 			lighting_build_overlays()
 		else
 			lighting_clear_overlays()
-	
+
 	obscured = old_obscured
 
 	return W
 
+/turf/proc/PreChange()
+	return
+
```

Comment:
I add this proc to answer the question of "What should a turf do before it is removed by `ChangeTurf`?"